awpy/stats/win_prob.py

```diff
@@ -0,0 +1,87 @@
+"""Calculates CT & T side likeliehood of winning a round at any given point"""
+
+from typing import List, Dict, Union
+
+from awpy import Demo
+
+
+def build_feature_matrix(demo: Demo, ticks: Union[int, List[int]]) -> List[Dict[str, float]]:
+    """
+    Builds the game state feature matrix for specified ticks.
+
+    Args:
+        demo (Demo): A parsed demo object containing CS2 match data.
+        ticks (Union[int, List[int]]): A single tick or a list of ticks at which to calculate features.
+
+    Returns:
+        List[Dict[str, Union[int, str, bool]]]: A list of dictionaries where each dictionary contains the game state features at a given tick.
+    
+    Raises:
+        ValueError: If specified ticks are not found within the demo object.
+    """
+    if not demo.ticks.empty:
+        if isinstance(ticks, int):
+            ticks = [ticks]
+        
+        # Gather round numbers for each specified tick
+        round_numbers = []
+        for tick in ticks:
+            tick_data = demo.ticks[demo.ticks["tick"] == tick]
+            if tick_data.empty:
+                raise ValueError(f"No data found for tick {tick}.")
+            round_numbers.append(tick_data["round"].iloc[0])
+        
+        # Check if all ticks belong to the same round
+        if len(set(round_numbers)) > 1:
```

Comment:
This is an interesting case. Should we throw an error? Nothing prevents us from predicting, so I would argue we don't even need to care about round numbers (also means we could axe the above code).

Reply:
I have removed this